### PR TITLE
add dash ability to soul, add new attack to dummy

### PR DESCRIPTION
**Pull request by emir4169**
funy

### CODE REVIEW

**Review by Dobby233Liu — COMMENTED** [review]
I think the line endings in scripts/world/maps/room1/data.lua and corrspending tmx got messed up

File: libraries/dash_soul/lib.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "id": "dashsoul",
+    "version": "1.0.0",
+    "engineVersion": "0.9.0-dev"
```
Comment (Dobby233Liu) [inline libraries/dash_soul/lib.json:4]:
```suggestion
    "version": "v1.0.0",
    "engineVersion": "v0.9.0-dev"
```

**Review by Dobby233Liu — CHANGES_REQUESTED** [review]
Not to mention we already have a room2, IDK how you handled it, it's hard to tell from the messed up diff

File: libraries/dash_soul/lib.json (hunk 1 above)
Comment (emir4169) [reply libraries/dash_soul/lib.json:4]: wait we already have one???

**Comment by emir4169** [comment]
(also please tell me if you know how to fix the great dash issue of not dashing)

**Comment by Dobby233Liu** [comment]
> (also please tell me if you know how to fix the great dash issue of not dashing)

Probably requires overwriting the doMovement function entirely. Makes this warrant becoming a separate object.

File: scripts/world/maps/room1/data.lua
Comment (Dobby233Liu) [inline scripts/world/maps/room1/data.lua:1]:
```bash
unix2dos scripts/world/maps/room1/data.lua
```

File: scripts/world/maps/room1/room1.tmx
Comment (Dobby233Liu) [inline scripts/world/maps/room1/room1.tmx:1]:
```bash
unix2dos scripts/world/maps/room1/room1.tmx
```

**Review by Dobby233Liu — CHANGES_REQUESTED** [review]

File: scripts/world/maps/room1/room1.tmx
Comment (Dobby233Liu) [reply scripts/world/maps/room1/room1.tmx:1]: If Tiled undos this it would be tragic

**Comment by Dobby233Liu** [comment]
> it's hard to tell from the messed up diff

Actually hiding whitespace changes did wonders here, aside of noise caused by inconsistent Tiled version, it seems you just touched some collision and added a loading zone for room2, so you didn't handle it at all. Suggest moving what you have in your room2 over to our room2 then remove the former

> (also please tell me if you know how to fix the great dash issue of not dashing)
I think it has something to do with the fact we already have a Soul hook file

**Comment by emir4169** [comment]
> > it's hard to tell from the messed up diff
> 
> Actually hiding whitespace changes did wonders here, aside from noise caused by inconsistent Tiled version, it seems you just touched some collision and added a loading zone for room2, so you didn't handle it at all. Suggest moving what you have in your room2 over to our room2, then remove the former
i shrimply just erased my room2 out of existence
> > (also please tell me if you know how to fix the great dash issue of not dashing)
> 
> I think it has something to do with the fact we already have a Soul hook file. We always had trouble with stuff like this
put the code in the soul hook file, doesnt seem to work still

**Comment by JustAnotherRandomGithubUser** [comment]
Closed due to a lack of activity.

**Pull request closed without merging**